README.md

```diff
@@ -276,26 +548,29 @@ The targets are searched for in following order.
 
 The workspace root is the first folder containing a `cabal.project` file.
 
-The arguments are collected by running `cabal v2-repl`.
+The arguments are collected by running `cabal v2-repl <filename>`.
 
-If `cabal v2-repl` fails, then the user needs to configure the correct
+If `cabal v2-repl <filename>` fails, then the user needs to configure the correct
 target to use by writing a `hie.yaml` file.
 
 ### `stack`
 
 The workspace root is the first folder containing a `stack.yaml` file.
 
-The arguments are collected by executing `stack repl`.
+The arguments are collected by executing `stack repl`. If it fails, then the user needs to configure the correct
+target to use by writing a `hie.yaml` file.
 
 ### `bios`
 
 The most general form is the `bios` mode which allows a user to specify themselves
 which flags to provide.
 
-In this mode, an executable file called `.hie-bios` is placed in the root
-of the workspace directory. The script takes one argument, the filepath
-to the current file we want to load into the session. The script returns
-a list of GHC arguments separated by newlines which will setup the correct session.
+The program will receive the file to return options for as its first argument.
+
+The process must consult the `HIE_BIOS_OUTPUT` environment variable and write a
+list of options to this file separated by newlines. Once the process finishes
```

Comment:
It is implicit and maybe obvious (with `this file`), but i would make clear that the value of `HIE_BIOS_OUTPUT` is the file path where you have to write the flags.

Reply:
I agree, we should make this clear!
Also, there is another undocumented pre-condition: all targets of the component must be listed by the process.
I will extend the documentation for the bios cradle.